src/blob_storage.cc

```diff
@@ -24,6 +24,26 @@ Status BlobStorage::NewPrefetcher(uint64_t file_number,
                                     result);
 }
 
+Status BlobStorage::DeleteBlobsInRanges(const RangePtr* ranges, size_t n,
+                                        bool include_end,
+                                        SequenceNumber obsolete_sequence) {
+  MutexLock l(&mutex_);
+  for (size_t i = 0; i < n; i++) {
+    auto begin = ranges[i].start, end = ranges[i].limit;
+    auto cmp = cf_options_.comparator;
+
+    for (auto it = blob_ranges_.lower_bound(*begin);
+         it != blob_ranges_.upper_bound(*end); it++) {
+      if (it->second->is_obsolete()) continue;
+      if ((include_end && cmp->Compare(it->second->largest_key(), *end) <= 0) ||
```

Comment:
don't we need to check if `largest_key()` is empty?

Reply:
good catch!